rest_server/src/models/marketplace_item.js

```diff
@@ -165,11 +161,57 @@ class MarketplaceItem {
             },
           ];
         }
-        const items = await this.orm.findAll({ where: filterStatement });
+        const havings = [];
+        if (tags) {
+          havings.push({
+            where: where(
+              fn('array_agg', col('ItemTags.name')),
+              Op.contains,
+              cast(tags, 'varchar[]'),
+            ),
+          });
```

Comment:
looks like the processing logic for Item and ItemCatogory are similiar, so how about create a function to do that so you don't need repeat yourself?

Reply:
Updated, add a new function and removed the duplicate code.